main.py

```diff
@@ -1,68 +1,63 @@
-import os
-import time
 import boto3
-import sys
+
 from watchdog.observers import Observer
 from watchdog.events import FileSystemEventHandler
   
-s3_bucket_name = ""
-filename = ""
-  
 class OnMyWatch:
     # Set the directory on watch
     watchDirectory = "e:/Code/indellinent/test/"
   
-    def __init__(self):
+    def __init__(self, handlers):
         self.observer = Observer()
-  
+        self.handlers = handlers
+
     def run(self):
-        event_handler = Handler()
-        self.observer.schedule(event_handler, self.watchDirectory)
+        for handler in self.handlers:
+            self.observer.schedule(handler, self.watchDirectory)
+
         self.observer.start()
         try:
-            while True:
-                time.sleep(5)
-        except:
+            while self.observer.isAlive():
+                self.observer.join(5)
+        except KeyboardInterrupt:
+            # user requested stop
+            pass
+        finally:
             self.observer.stop()
+            self.observer.join()
             print("Observer Stopped")
-  
-        self.observer.join()
-  
-  
+
 class Handler(FileSystemEventHandler):
```

Comment:
Just remembered that boto3 is using some kind of dynamic type stuff (it's been a couple of years) so usually I had to use `# type: ignore` to pacify mypy and move on.

On the up-side, using `# type: ignore` would indicate to your reviewers that you know about mypy and assume it will be used and (hopefully) are even using it yourself.